test/test_build_python.py

```diff
@@ -45,60 +45,101 @@ def monkey_patch_put_event_into_queue(monkeypatch):
-            tmp_path = Path(tmp_path_str)
-            python_build_task = PythonBuildTask()
-            package = PackageDescriptor(tmp_path / 'src')
-            package.name = 'test_package'
-            package.type = 'python'
+        tmp_path = Path(tmp_path_str)
+        python_build_task = PythonBuildTask()
+        package = PackageDescriptor(tmp_path / 'src')
+        package.name = 'test_package'
+        package.type = 'python'
+        package.metadata['get_python_setup_options'] = lambda _: {
+            'packages': ['my_module'],
+        }
 
-            context = TaskContext(
-                pkg=package,
-                args=SimpleNamespace(
-                    path=str(tmp_path / 'src'),
-                    build_base=str(tmp_path / 'build'),
-                    install_base=str(tmp_path / 'install'),
-                    symlink_install=False,
-                ),
-                dependencies={}
-            )
-            python_build_task.set_context(context=context)
+        context = TaskContext(
+            pkg=package,
+            args=SimpleNamespace(
+                path=str(tmp_path / 'src'),
+                build_base=str(tmp_path / 'build'),
+                install_base=str(tmp_path / 'install'),
+                symlink_install=symlink_install,
+            ),
+            dependencies={}
+        )
+        python_build_task.set_context(context=context)
+
+        pkg = python_build_task.context.pkg
+
+        pkg.path.mkdir(exist_ok=True)
+        (pkg.path / 'setup.py').write_text(
+            'from setuptools import setup\n'
+            'setup(\n'
+            '    name="test_package",\n'
+            '    packages=["my_module"],\n'
+            ')\n'
+        )
+        (pkg.path / 'my_module').mkdir(exist_ok=True)
+        (pkg.path / 'my_module' / '__init__.py').touch()
+
+        src_base = Path(python_build_task.context.args.path)
+
+        source_files_before = set(src_base.rglob('*'))
+        rc = event_loop.run_until_complete(python_build_task.build())
+        assert not rc
+        source_files_after = set(src_base.rglob('*'))
+        assert source_files_before == source_files_after
+
+        build_base = Path(python_build_task.context.args.build_base)
+        assert build_base.rglob('my_module/__init__.py')
+
+        return Path(python_build_task.context.args.install_base)
+    finally:
+        event_loop.close()
 
-            pkg = python_build_task.context.pkg
 
-            pkg.path.mkdir()
-            (pkg.path / 'setup.py').write_text(
-                'from setuptools import setup\n'
-                'setup(\n'
-                '    name="test_package",\n'
-                '    packages=["my_module"],\n'
-                ')\n'
-            )
-            (pkg.path / 'my_module').mkdir()
-            (pkg.path / 'my_module' / '__init__.py').touch()
+def test_build_package():
+    with TemporaryDirectory(prefix='test_colcon_') as tmp_path_str:
```

Comment:
The way this particular context manager works, it is indeed a string: https://github.com/python/cpython/blob/c829975428253568d47ebfc3104fa7386b5e0b58/Lib/tempfile.py#L914-L915

I specifically suffixed it `_str` because I'm immediately converting it to a `Path` instance called `tmp_path` later.

Thanks for having a look!